src/workerd/api/tests/http-socket-test.js

```diff
@@ -0,0 +1,247 @@
+
+      const response = await httpClient.fetch('/ping');
+      assert.equal(response.status, 200);
+      const text = await response.text();
+      assert.equal(text, 'pong');
+    }
+
+    // Test JSON response
+    {
+      const socket = connect(`localhost:${env.HTTP_SOCKET_SERVER_PORT}`);
+      const httpClient = convertSocketToFetcher(socket);
+
+      const response = await httpClient.fetch('/json');
+      assert.equal(response.status, 200);
+      assert.equal(response.headers.get('content-type'), 'application/json');
+      const data = await response.json();
+      assert.deepEqual(data, { message: 'Hello from HTTP socket server' });
+    }
+
+    // Test POST request with body
+    {
+      const socket = connect(`localhost:${env.HTTP_SOCKET_SERVER_PORT}`);
+      const httpClient = convertSocketToFetcher(socket);
+
+      const postData = 'Hello, world!';
+      const response = await httpClient.fetch('/echo', {
+        method: 'POST',
+        body: postData,
+      });
+      assert.equal(response.status, 200);
+      const text = await response.text();
+      assert.equal(text, postData);
+    }
+
+    // Test request with custom headers
+    {
+      const socket = connect(`localhost:${env.HTTP_SOCKET_SERVER_PORT}`);
+      const httpClient = convertSocketToFetcher(socket);
+
+      const response = await httpClient.fetch('/headers', {
+        headers: {
+          'X-Custom-Header': 'custom-value',
+          'X-Another-Header': 'another-value',
+        },
+      });
+      assert.equal(response.status, 200);
+      const headers = await response.json();
+      assert.equal(headers['x-custom-header'], 'custom-value');
+      assert.equal(headers['x-another-header'], 'another-value');
+    }
+
+    // Test 404 response
+    {
+      const socket = connect(`localhost:${env.HTTP_SOCKET_SERVER_PORT}`);
+      const httpClient = convertSocketToFetcher(socket);
+
+      const response = await httpClient.fetch('/status/404');
+      assert.equal(response.status, 404);
+      const text = await response.text();
+      assert.equal(text, 'Not Found');
+    }
+
+    // Test 500 response
+    {
+      const socket = connect(`localhost:${env.HTTP_SOCKET_SERVER_PORT}`);
+      const httpClient = convertSocketToFetcher(socket);
+
+      const response = await httpClient.fetch('/status/500');
+      assert.equal(response.status, 500);
+      const text = await response.text();
+      assert.equal(text, 'Internal Server Error');
+    }
+
+    // Test multiple requests on same connection
+    {
+      const socket = connect(`localhost:${env.HTTP_SOCKET_SERVER_PORT}`);
+      const httpClient = convertSocketToFetcher(socket);
+
+      // First request
+      const response1 = await httpClient.fetch('/ping');
+      assert.equal(response1.status, 200);
+      const text1 = await response1.text();
+      assert.equal(text1, 'pong');
+      try {
+        // Second request on same connection
```

Comment:
This test is confusing. Are you expecting this to throw or succeed? If you are expecting a throw, using the `node:assert` `throws(...)` API would be a better way to construct this block